src/Activator.php

```diff
@@ -27,5 +29,33 @@ protected function hooks() {
 	 */
 	public static function run(): void {
 		Campaigner::register_role_and_add_capabilities();
+		self::do_migrations();
```

Comment:
I started with a separate object, but then realized that this Activator is actually already performing a migration, as it updates the capabilities in the db. Note that the class already did more than register the theme switch hook. The same function is also run through the command `p4-run-activator`. So yeah, we definitely want to clean this up a bit, I'll see what I can make of it :+1:

Reply:
I created a `Migrator` class instead to hold this logic. Pretty much the same as the `MigrationRunner` from your branch, but since it's a class with no members and a single function I made it a static function. We can still change it to an instance method and create `Migrator` instances if we would need that instance to hold state or dependencies, just always doing that in advance will lead to a lot of unused instances hanging around. Basically if you find a class being used this way 
``` ( new MyClass())->doFoo() ```, it de facto does not need the instance.

Reply:
> Basically if you find a class being used this way `( new MyClass())->doFoo()`, it de facto does not need the instance.

We will disagree on this for many years to come 😁 but it doesn't matter too much here 👍 ☮️

Reply:
I mean this in a more neutral way :peace_symbol: : in that `de facto` there will be an instance which is created and completely not used. There is no input in the constructor so any members that instance would have would be static values (unless of course the constructor is pulling things from the global scope, which should definitely not happen in the constructor).